src/daft-core/src/series/ops/sign.rs

```diff
@@ -0,0 +1,24 @@
+use crate::datatypes::DataType;
+use crate::series::Series;
+use common_error::DaftError;
+use common_error::DaftResult;
+
+impl Series {
+    pub fn sign(&self) -> DaftResult<Series> {
+        use crate::series::array_impl::IntoSeries;
+        use DataType::*;
+        match self.data_type() {
+            UInt8 | UInt16 | UInt32 | UInt64 => Ok(self.clone()),
```

Comment:
Need to call `.sign()?` on these as well otherwise the value is returned as is.

Reply:
is there any way i can trigger this condition from python test ?

Reply:
Absolutely! You can create a MicroPartition from a pyarrow table with uint columns. For example:
```
table = MicroPartition.from_arrow(
    pa.Table(
        {
            "uint8": pa.array([None, 0, 1, 2, 3], type=pa.uint8()),
        }
    )
)
```

Reply:
is there no way to 
```impl<T: DaftNumericType> DataArray<T>
where T::Native: Unsigned
{
    pub fn sign(&self) -> DaftResult<Self> {
        self.apply(|v| if v as u64  == 0 { 0 } else { 1 }
)
    }
}
```

currently i have implemented it individually

Reply:
You can use the zero and one identities from numtraits https://docs.rs/num-traits/0.2.18/num_traits/identities/index.html
```
impl<T: DaftNumericType> DataArray<T>
where
    T::Native: Unsigned,
{
    pub fn sign_unsigned(&self) -> DaftResult<Self> {
        self.apply(|v| {
            if v.is_zero() {
                T::Native::zero()
            } else {
                T::Native::one()
            }
        })
    }
}
```

Reply:
ok type casting is hard...
also if i need to do something like v * 2 .
then do i need to impl Int8Array then do operation or i can do it in DaftNumericType

Reply:
Indeed it is haha, for arithmetic operations we use a different trait bound, you can check it out here: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/arithmetic.rs#L109